### PR TITLE
Documentation - Basic&Prehash

### LINKED ISSUE #70: Documentation (#51) - Basic & Prehashed construction
## Why

User-friendly explanation of the Telescope scheme.

## What
- [ ] #84 
- [ ] #85 

Linked to issue #51

### PR DESCRIPTION
**Pull request by curiecrypt**
## Content
Cargo documentation for 
- Telescope basic construction
- Telescope construction with pre-hashing.

## Pre-submit checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
- PR
    - [x] No clippy warnings in the CI
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- Documentation
    - [x] Update documentation website (if relevant)

## Issue(s)

Closes #70

### CODE REVIEW

File: docs/basic_construction.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+>   - **if** $~~ \mathsf{H_1}(t, slist_{new}) = 1:$
+>     - $\pi \leftarrow \mathsf{DFS}(t, slist_{new}, S_p)$
+>     - **if** $~~ \pi ~=\not \emptyset:$
+>       - **return** $~~ \pi.$
+> - **return** $~~ \emptyset.$
+---
+
+#### Verification
+---
+> $\mathsf{Verify}(\pi) \leftarrow 0/1$
```
Comment (tolikzinovyev) [inline docs/basic_construction.md:112]:
```suggestion
> $\mathsf{Verify}(\pi) \rightarrow 0/1$
```

File: docs/basic_construction.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+> 
+> - Input:
+>   - $\pi:~~$ `proof`, a proof of the form $(t, s_1, \ldots, s_u)$.
+> - Output:
+>   - $0/1$.
+> ---
+> - **if** $~~ t \in\not  \[d\]:$
```
Comment (tolikzinovyev) [inline docs/basic_construction.md:119]: this also doesn't render well for me

File: docs/basic_construction.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+>   - **return** $~~ 0.$
+> - **for** each $~~ i = \[1,~ u\]:$
+>   - **if** $~~ \mathsf{H_1}(t, s_1, \ldots, s_i) ~=\not 1:$
```
Comment (tolikzinovyev) [inline docs/basic_construction.md:122]: and this

**Review by tolikzinovyev — COMMENTED** [review]
Reviewed the basic construction, it looks good I think! Left comments about some inaccuracies and rendering issues. Will defer stylistic reviews to @rrtoledo and @djetchev. :)

Do you think we could have prehashing in a separate PR? :)  Suggest merging basic construction first and then writing prehashing in a similar agreed-upon style.

File: docs/basic_construction.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
+# Telescope - Basic Construction
```
[the remark this answers is not in the document]
Comment (curiecrypt) [reply docs/basic_construction.md:1]: I plan to wrap up the structure of the documentation at the end of the `intro` page. 
I will put a paragraph explaining 
- the content of each section
- the logic behind this organization
- flow of information
- ...
We can mention these kind of things in that paragraph. However, the documentation is very tiny now. So, it is not possible to write it yet.

**Pull request closed without merging**